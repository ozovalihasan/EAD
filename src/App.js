import React, { useState } from 'react';
import styled from 'styled-components';
import { DragDropContext } from 'react-beautiful-dnd';
import initialData from './initial-data';
import Column from './column';

const App = () => {
  // setTimeout(() => { console.warn('hasan'); }, 0);
  const [data, setData] = useState(initialData);

  // const onDragStart = () => {
  //   document.body.style.color = 'orange';
  //   document.body.style.transition = 'background-color 0.2s ease';
  // };

  // const onDragUpdate = (update) => {
  //   const { destination } = update;
  //   const opacity = destination
  //     ? destination.index / Object.keys(data.tasks).length
  //     : 0;
  //   document.body.style.backgroundColor = `rgba(153, 141, 217, ${opacity})`;
  // };
  const onDragStart = (start) => {
    const homeIndex = data.columnOrder.indexOf(start.source.droppableId);

    setData({
      ...data,
      homeIndex,
    });
  };
  const onDragEnd = (result) => {
    const {
      destination, source, draggableId,
    } = result;

    setData({
      ...data,
      homeIndex: null,
    });
    if (!destination) {
      return;
    }

    if (
      destination.droppableId === source.droppableId
      && destination.index === source.index
    ) {
      return;
    }

    const home = data.columns[source.droppableId];
    const foreign = data.columns[destination.droppableId];

    if (home === foreign) {
      const newTaskIds = Array.from(home.taskIds);
      newTaskIds.splice(source.index, 1);
      newTaskIds.splice(destination.index, 0, draggableId);

      const newHome = {
        ...home,
        taskIds: newTaskIds,
      };

      const newData = {
        ...data,
        columns: {
          ...data.columns,
          [newHome.id]: newHome,
        },
      };

      setData(newData);
      return;
    }

    const homeTaskIds = Array.from(home.taskIds);
    homeTaskIds.splice(source.index, 1);
    const newHome = {
      ...home,
      taskIds: homeTaskIds,
    };

    const foreignTaskIds = Array.from(foreign.taskIds);
    foreignTaskIds.splice(destination.index, 0, draggableId);
    const newForeign = {
      ...foreign,
      taskIds: foreignTaskIds,
    };

    const newData = {
      ...data,
      columns: {
        ...data.columns,
        [newHome.id]: newHome,
        [newForeign.id]: newForeign,
      },
    };
    setData(newData);
  };

  return (
    <div className="App">
      <DragDropContext
        onDragStart={onDragStart}
        onDragEnd={onDragEnd}
      >
        <Container>

          {data.columnOrder.map((columnId, index) => {
            const column = data.columns[columnId];
            const tasks = column.taskIds.map(
              (taskId) => data.tasks[taskId],
            );
            const isDropDisabled = index < data.homeIndex;

            return (
              <Column
                key={column.id}
                column={column}
                tasks={tasks}
                isDropDisabled={isDropDisabled}
              />
            );
          })}

        </Container>

      </DragDropContext>

    </div>
  );
};

const Container = styled.div`
  display: flex;
`;

export default App;
