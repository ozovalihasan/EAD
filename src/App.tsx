import React, { useState, useRef, useCallback } from 'react';
import ReactFlow, {
  ReactFlowProvider,
  Controls,
  Node,
  ReactFlowInstance,
} from 'react-flow-renderer';

import Sidebar from './Sidebar';
import './index.css';
import EntityNode, { EntityNodeDataType } from './EntityNode';
import useStore from './zustand/store';
import HasManyEdge from './HasManyEdge';
import HasOneEdge from './HasOneEdge';
import ThroughEdge from './ThroughEdge';
import ConnectionLine from './ConnectionLine';



const nodeTypes = {
  entity: EntityNode,
};
const edgeTypes = {
  hasMany: HasManyEdge,
  hasOne: HasOneEdge,
  through: ThroughEdge,
};

const App = () => {
  
  const { 
    nodes, 
    edges, 
    onNodesChange, 
    onEdgesChange, 
    onConnect, 
    addNode, 
    onConnectStart, 
    onConnectEnd, 
    onNodeMouseEnter, 
    onNodeMouseLeave,
    onEdgeMouseEnter, 
    onEdgeMouseLeave,
  } = useStore();
  const reactFlowWrapper = useRef<HTMLDivElement>(null);
  const [reactFlowInstance, setReactFlowInstance] = useState<ReactFlowInstance<any, any> | null>(null);


  const onDragOver = useCallback((event: React.DragEvent<HTMLDivElement>) => {
    event.preventDefault();
    event.dataTransfer.dropEffect = 'move';
  }, []);

  const onDrop = useCallback(
    (event: React.DragEvent<HTMLDivElement>) => {
      event.preventDefault();

      let reactFlowBounds: DOMRect = (reactFlowWrapper.current as HTMLDivElement).getBoundingClientRect();
      
      const type = event.dataTransfer.getData('application/reactflow');
      const tableId = event.dataTransfer.getData('tableId');

      // check if the dropped element is valid
      if (typeof type === 'undefined' || !type) {
        return;
      }

      let position: {x: number, y: number} = (reactFlowInstance as ReactFlowInstance).project({
          x: event.clientX - reactFlowBounds.left,
          y: event.clientY - reactFlowBounds.top,
        });

      
      let name = useStore.getState().tables[tableId].name;

      const newNode: Node<EntityNodeDataType> = {
        id: useStore.getState().idCounter.toString(),
        type: "entity",
        position,
        data: { tableId, name },
      };
      

      addNode(newNode);
    },
    [reactFlowInstance]
  );


  return (
    <div  className="h-screen w-screen bg-first-100 flex flex-row flex-grow">
      <ReactFlowProvider>
        <div className="h-full flex-grow" ref={reactFlowWrapper}>

          <ReactFlow
            nodes={nodes}
            edges={edges}
            onNodesChange={onNodesChange}
            onEdgesChange={onEdgesChange}
            onConnect={onConnect}
            onInit={setReactFlowInstance}
            onDrop={onDrop}
            onConnectStart={onConnectStart}
            onConnectEnd={onConnectEnd}
            onNodeMouseEnter={onNodeMouseEnter}
            onNodeMouseLeave={onNodeMouseLeave}
            onEdgeMouseEnter={onEdgeMouseEnter}
            onEdgeMouseLeave={onEdgeMouseLeave}
            onDragOver={onDragOver}
            edgeTypes={edgeTypes}
            deleteKeyCode={"Delete"}
            connectionLineComponent={ConnectionLine }
            elevateEdgesOnSelect={true}
            fitView
            nodeTypes={nodeTypes}
            snapToGrid={true}
            
          >
            <Controls />

          </ReactFlow>
        </div>
        <Sidebar />
      </ReactFlowProvider>
    </div>

  )
}

export default App
