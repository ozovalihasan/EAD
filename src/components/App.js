import React from 'react';
import { DragDropContext, Droppable } from 'react-beautiful-dnd';
import { useDispatch, useSelector } from 'react-redux';
import { library } from '@fortawesome/fontawesome-svg-core';
import {
  faArrowsAlt, faExpandAlt, faCompressAlt, faEllipsisH, faEllipsisV, faFlag, faClone,
  faPlus, faPlaneDeparture, faPlaneArrival, faTimes,
} from '@fortawesome/free-solid-svg-icons';
import { faGithub } from '@fortawesome/free-brands-svg-icons';
import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Model from './Model';
import {
  removeItem,
  addItem,
  moveItem,
  updateDraggedItemId,
  resetState,
  installState,
  idCountIncrease,
  toggleExpandAll,
  toggleCompactMode,
  updateRestrictedDropId,
} from '../redux';
import saveJSON from './saveJSON';
import colors from './colors';

library.add(faArrowsAlt, faExpandAlt, faCompressAlt, faEllipsisH,
  faEllipsisV, faFlag, faClone, faGithub, faPlus, faPlaneDeparture, faPlaneArrival, faTimes);

const App = () => {
  const {
    items, dragDropCategory, idCount, expandAll, compactMode,
  } = useSelector((state) => state.block);

  const startingId = compactMode ? 9 : 0;
  const dispatch = useDispatch();

  const checkDragDropCategory = (dragId, dropId) => (
    dragDropCategory[items[dragId].category].includes(items[dropId].category)
  );

  const stateBlock = useSelector((state) => state.block);

  const saveBlocks = () => {
    localStorage.block = JSON.stringify(stateBlock);
  };

  const onDragStart = (result) => {
    dispatch(updateDraggedItemId(result.draggableId));
  };

  const onDragEnd = (result) => {
    dispatch(updateDraggedItemId(-1));
    const {
      destination, draggableId, source,
    } = result;

    if (!destination) {
      if (!(items[draggableId].factory)) {
        dispatch(removeItem(source.droppableId, source.index, draggableId));
      }
      return;
    }

    if (!checkDragDropCategory(draggableId, destination.droppableId)) {
      return;
    }

    if (items[draggableId].factory) {
      dispatch(addItem(draggableId, destination.droppableId, destination.index, idCount));
      if (items[draggableId].association) {
        dispatch(updateRestrictedDropId(idCount, null));
      }
      dispatch(idCountIncrease());
      return;
    }

    if (items[destination.droppableId].entityContainer
      && items[destination.droppableId].subItemIds.length > 0
      && (items[items[destination.droppableId].subItemIds[0]].category
        !== items[draggableId].category
      )) {
      return;
    }
    dispatch(moveItem(
      draggableId,
      destination.droppableId,
      destination.index,
      source.droppableId,
      source.index,
    ));
  };

  return (
    <MainContainer className="App">
      <LogoButtons>
        <Logo src={`${process.env.PUBLIC_URL}/images/ead-logo.svg`} alt="EAD logo" />
        <Button
          onClick={() => saveJSON(items, 'EAD.json')}
          type="button"
        >
          Download EAD
        </Button>

        <CompactMode
          onClick={() => dispatch(toggleCompactMode())}
          type="button"
          compactMode={compactMode}
        >
          Compact Mode
        </CompactMode>

        {compactMode
        || (
          <>
            <Button
              onClick={() => dispatch(resetState())}
              type="button"
            >
              Reset
            </Button>
            <Button
              onClick={() => saveBlocks()}
              type="button"
            >
              Save
            </Button>
            <Button
              onClick={() => dispatch(installState())}
              type="button"
            >
              Install Saved Data
            </Button>

            <ExpandAllButton
              onClick={() => dispatch(toggleExpandAll())}
              type="button"
              expandAll={expandAll}
            >
              Expand All
            </ExpandAllButton>

            <GithubLink href="https://github.com/ozovalihasan/ead">
              <FontAwesomeIcon icon={faGithub} size="2x" />
            </GithubLink>
          </>
        )}
      </LogoButtons>
      <DragDropContext
        onDragStart={onDragStart}
        onDragEnd={onDragEnd}
      >
        <Droppable
          droppableId="OuterDrop"
          direction="vertical"
          isDropDisabled
        >

          { (provided) => (
            <div
              {...provided.droppableProps}
              ref={provided.innerRef}
            >
              <Model
                parentId={startingId}
                item={items[startingId]}
                allItems={items}
                index={startingId}
                checkDragDropCategory={checkDragDropCategory}
              />
              {provided.placeholder}
            </div>
          )}

        </Droppable>
      </DragDropContext>
    </MainContainer>
  );
};

const MainContainer = styled.div`
  font-family: Arial, Helvetica, sans-serif;
`;

const LogoButtons = styled.div`
  display: flex;
  width: 100%;
  align-items: center;
`;

const Logo = styled.img`
  width: 60px;
  padding-right: 40px;
`;

const Button = styled.button`
  border-radius: 5px;
  border: 1px solid ${colors.disabled};
  background-color: transparent;
  margin: 10px;
  padding: 10px;
  transition: background-color 0.2s;

  &:hover {
    cursor: pointer;
    background-color: ${colors.entity};
  }
`;

const ExpandAllButton = styled(Button)`
  background-color: ${(props) => (props.expandAll && colors.association)};
`;

const CompactMode = styled(Button)`
  background-color: ${(props) => (props.compactMode && colors.entity)};
`;

const GithubLink = styled.a`
  margin: 0 10px;
  color: #000;

  &:hover {
    color: ${colors.chosen};
  }
`;
export default App;
