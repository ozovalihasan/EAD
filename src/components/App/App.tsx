import {
  Sidebar, 
  Navbar,
  SidebarHandle,
  MainReactFlow
} from "@/components"
import { useCustomizationStore } from '@/zustandStore/customizationStore';

export const App = () => {
  
  const locationSidebar = useCustomizationStore((state) => state.locationSidebar)
  
  return (

    <div className="text-zinc-600 dark:text-zinc-100 font-default bg-zinc-50 dark:bg-zinc-700 flex flex-col h-screen">
      <Navbar />
      <div  className={`h-[calc(100vh-5rem)] w-screen  flex flex-grow ${locationSidebar === "left" ? "flex-row" : "flex-row-reverse"}`}>
        <Sidebar />
        <SidebarHandle />
        <MainReactFlow />
      </div>
    </div>

  )
}