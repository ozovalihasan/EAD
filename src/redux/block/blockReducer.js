/* eslint-disable no-param-reassign */
import { createSlice } from '@reduxjs/toolkit';

const blockSlice = createSlice({
  name: 'block',
  initialState: {
    items: {
      0: {
        content: 'Initialize your project from ERD',
        subItemIds: [1, 8],
        order: 'vertical',
        subdirection: 'row',
        isDropDisabled: true,
        isDragDisabled: true,
      },
      1: {
        content: 'Elements',
        subItemIds: [2, 3, 4, 5, 6, 7],
        order: 'vertical',
        subdirection: 'column',
        isDropDisabled: true,
        isDragDisabled: true,
      },
      2: {
        content: 'has_many',
        subItemIds: [],
        order: 'vertical',
        subdirection: 'column',
        isDropDisabled: true,
        factory: true,
        association: true,
      },
      3: {
        content: 'has_one',
        subItemIds: [],
        order: 'vertical',
        subdirection: 'column',
        isDropDisabled: true,
        factory: true,
        association: true,
      },
      4: {
        content: 'belongs_to',
        subItemIds: [],
        order: 'vertical',
        subdirection: 'column',
        isDropDisabled: true,
        factory: true,
        association: true,
      },
      5: {
        content: 'has_and_belongs_to_many',
        subItemIds: [],
        order: 'vertical',
        subdirection: 'column',
        isDropDisabled: true,
        factory: true,
        association: true,
      },
      6: {
        content: 'attribute',
        subItemIds: [],
        attribute: true,
        order: 'vertical',
        subdirection: 'column',
        isDropDisabled: true,
        factory: true,
        type: 'string',
      },
      7: {
        content: 'entity',
        subItemIds: [],
        order: 'vertical',
        subdirection: 'column',
        factory: true,
        entity: true,
      },
      8: {
        content: 'ERD',
        subItemIds: [9],
        order: 'vertical',
        subdirection: 'column',
        isDragDisabled: true,
      },
      9: {
        content: 'entity',
        subItemIds: [],
        order: 'vertical',
        subdirection: 'column',
        factory: false,
        entity: true,
        isDropDisabled: false,
      },
    },
    restrictedDropId: -1,
  },
  reducers: {
    changeType: {
      reducer: (state, { payload }) => {
        state.items[payload.id].type = payload.value;
      },
      prepare: (e, id) => {
        const { value } = e.target;
        return { payload: { value, id } };
      },
    },

    changeContent: {
      reducer: (state, { payload }) => {
        state.items[payload.id].content = payload.value;
      },
      prepare: (e, id) => {
        const { value } = e.target;
        return { payload: { value, id } };
      },
    },

    checkDirection: {
      reducer: (state, { payload }) => {
        state.items[payload.id].subdirection = payload.subdirection;
        state.items[payload.id].order = payload.order;
      },
      prepare: (id, subdirection, order) => ({
        payload: { id, subdirection, order },
      }),
    },

    removeItem: {
      reducer: (state, { payload }) => {
        state.items[payload.parentId].subItemIds.splice(payload.itemIndexInSource, 1);
        delete state.items[payload.itemId];
      },
      prepare: (parentId, itemIndexInSource, itemId) => (
        {
          payload: {
            parentId, itemIndexInSource, itemId,
          },
        }
      ),
    },

    addItem: {
      reducer: (state, { payload }) => {
        const newItem = {
          ...state.items[payload.itemId],
          subItemIds: [],
          factory: false,
          isDropDisabled: false,
        };
        state.items[payload.newId] = newItem;
        state.items[payload.containerId].subItemIds.splice(
          payload.newId, 0, payload.newId,
        );
      },
      prepare: (itemId, containerId, containerIndex, newId) => (
        {
          payload: {
            itemId, containerId, containerIndex, newId,
          },
        }
      ),
    },

    moveItem: {
      reducer: (state, { payload }) => {
        state.items[payload.prevContainerId].subItemIds.splice(payload.prevContainerIndex, 1);
        state.items[payload.containerId].subItemIds.splice(
          payload.containerIndex, 0, parseInt(payload.itemId, 10),
        );
      },
      prepare: (itemId, containerId, containerIndex, prevContainerId, prevContainerIndex) => (
        {
          payload: {
            itemId, containerId, containerIndex, prevContainerId, prevContainerIndex,
          },
        }
      ),
    },
    updateRestrictedDropId: {
      reducer: (state, { payload }) => {
        state.restrictedDropId = payload.itemId;
      },
      prepare: (itemId) => (
        {
          payload: {
            itemId,
          },
        }
      ),
    },

  },

});

const { actions, reducer } = blockSlice;

export const {
  changeType,
  changeContent,
  checkDirection,
  removeItem,
  addItem,
  moveItem,
  updateRestrictedDropId,
} = actions;

export default reducer;
