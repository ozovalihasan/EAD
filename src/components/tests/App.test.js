/* eslint-disable global-require */
import { configureStore } from '@reduxjs/toolkit';
import { screen, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';

import userEvent from '@testing-library/user-event';
import { Simulate } from 'react-dom/test-utils';
import blockReducer from '../../redux/block/blockReducer';
import App from '../App';

jest.mock('../saveJSON', () => jest.fn());
const saveJSONClick = require('../saveJSON');

jest.mock('../../redux/block/initialState', () => {
  const testInitialState = {
    items: { 0: { content: 'mock content' } },
    dragDropCategory: { },
  };
  return testInitialState;
});

jest.mock('../Model', () => {
  const PropTypes = require('prop-types');
  const MockModel = ({
    item,
    allItems,
    checkDragDropCategory,
  }) => (
    <div>
      Mock Model
      {JSON.stringify(allItems)}
      hasan
      {JSON.stringify(item)}
      hasan
      {JSON.stringify(checkDragDropCategory)}
    </div>
  );

  MockModel.propTypes = {
    item: PropTypes.shape().isRequired,
    allItems: PropTypes.shape().isRequired,
    checkDragDropCategory: PropTypes.func.isRequired,
  };
  MockModel.displayName = 'MockModel';
  return MockModel;
});

const localStorageMock = {};
Object.defineProperty(window, 'localStorage', { value: localStorageMock });

const createTestStore = () => {
  const testStore = configureStore({
    reducer: { block: blockReducer },
    middleware: (getDefaultMiddleware) => getDefaultMiddleware(),
    devTools: false,
  });
  return testStore;
};

let store;
let renderReadyComponent;
describe('<App />', () => {
  beforeEach(() => {
    store = createTestStore();
    store.dispatch = jest.fn();

    renderReadyComponent = (
      <Provider store={store}>
        <App />
      </Provider>
    );
  });

  it('renders Model component', () => {
    render(renderReadyComponent);

    expect(screen.getByText(/Mock Model/i)).toBeInTheDocument();
    expect(screen.getByText(/{"0":{"content":"mock content"}}/i)).toBeInTheDocument();
    expect(screen.getByText(/{"content":"mock content"}/i)).toBeInTheDocument();
  });

  it('calls saveJSON function if \'Download EAD\' button is clicked', () => {
    render(renderReadyComponent);
    userEvent.click(screen.getByText(/Download EAD/i));

    expect(saveJSONClick.mock.calls.length).toBe(1);
  });

  it('dispatches toggleCompactMode action if \'Compact Mode\' button is clicked', () => {
    render(renderReadyComponent);
    userEvent.click(screen.getByText('Compact Mode'));

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch.mock.calls[0][0].type).toBe('block/toggleCompactMode');
  });

  it('dispatches toggleCompactMode action if \'More Text\' button is clicked', () => {
    render(renderReadyComponent);
    userEvent.click(screen.getByText('More Text'));

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch.mock.calls[0][0].type).toBe('block/toggleMoreText');
  });

  it('dispatches uploadAllData action if \'Browse\' button is clicked and a file is selected', () => {
    const readAsText = jest.fn();
    const fileReader = {
      readAsText,
    };

    jest.spyOn(window, 'FileReader').mockImplementation(() => fileReader);
    render(renderReadyComponent);
    Simulate.change(screen.getByTestId('uploadInput'), { target: { files: [new Blob()] } });

    expect(readAsText.mock.calls.length).toBe(1);

    fileReader.onload({ target: { result: JSON.stringify('result') } });

    expect(store.dispatch.mock.calls[0][0].type).toBe('block/uploadAllData');
  });

  it('dispatches resetState action if \'Reset\' button is clicked', () => {
    render(renderReadyComponent);
    userEvent.click(screen.getByText('Reset'));

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch.mock.calls[0][0].type).toBe('block/resetState');
  });

  it('save state to local storage if \'Save\' button is clicked', () => {
    render(renderReadyComponent);
    userEvent.click(screen.getByText('Save'));

    expect(localStorageMock.block === JSON.stringify(store.getState().block)).toBe(true);
  });

  it('dispatches installState action if \'Install Saved Data\' button is clicked', () => {
    render(renderReadyComponent);
    userEvent.click(screen.getByText('Install Saved Data'));

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch.mock.calls[0][0].type).toBe('block/installState');
  });

  it('dispatches toggleExpandAll action if \'Expand All\' button is clicked', () => {
    render(renderReadyComponent);
    userEvent.click(screen.getByText('Expand All'));

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch.mock.calls[0][0].type).toBe('block/toggleExpandAll');
  });

  it('renders a link to the repository of the project', () => {
    render(renderReadyComponent);
    expect(screen.getByTitle('Click to see the repository of the project').closest('a')).toHaveAttribute('href', 'https://github.com/ozovalihasan/ead');
  });

  it('renders correctly', () => {
    const renderedContainer = render(renderReadyComponent);

    expect(renderedContainer).toMatchSnapshot();
  });
});
