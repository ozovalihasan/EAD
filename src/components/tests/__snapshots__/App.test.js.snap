// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blockReducer <App /> renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-0"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-1"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-2"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-3"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-4"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-5"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div>
      <div
        class="sc-bdvvaa bqALdu App"
      >
        <button
          class="sc-gsDJrp qnnxF"
          type="button"
        >
          Download EAD
        </button>
        <button
          class="sc-gsDJrp qnnxF"
          type="button"
        >
          Reset
        </button>
        <button
          class="sc-gsDJrp qnnxF"
          type="button"
        >
          Save
        </button>
        <button
          class="sc-gsDJrp qnnxF"
          type="button"
        >
          Install Saved Data
        </button>
        <button
          class="sc-gsDJrp sc-dkPtyc qnnxF bDqtzn"
          type="button"
        >
          Expand All
        </button>
        <div
          data-rbd-droppable-context-id="6"
          data-rbd-droppable-id="OuterDrop"
        >
          <div>
            Mock Model
            {"0":{"content":"mock content"}}
            hasan
            {"content":"mock content"}
            hasan
          </div>
        </div>
      </div>
    </div>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-6"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      id="rbd-hidden-text-6-hidden-text-12"
      style="display: none;"
    >
      
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvaa bqALdu App"
    >
      <button
        class="sc-gsDJrp qnnxF"
        type="button"
      >
        Download EAD
      </button>
      <button
        class="sc-gsDJrp qnnxF"
        type="button"
      >
        Reset
      </button>
      <button
        class="sc-gsDJrp qnnxF"
        type="button"
      >
        Save
      </button>
      <button
        class="sc-gsDJrp qnnxF"
        type="button"
      >
        Install Saved Data
      </button>
      <button
        class="sc-gsDJrp sc-dkPtyc qnnxF bDqtzn"
        type="button"
      >
        Expand All
      </button>
      <div
        data-rbd-droppable-context-id="6"
        data-rbd-droppable-id="OuterDrop"
      >
        <div>
          Mock Model
          {"0":{"content":"mock content"}}
          hasan
          {"content":"mock content"}
          hasan
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
